integrations/tensorflow/bindings/python/pyiree/tf/support/tf_test_utils.py

```diff
@@ -55,9 +55,14 @@
 
 
 def _setup_artifacts_dir(module_name: str) -> str:
-  parent_dir = FLAGS.artifacts_dir
-  if parent_dir is None:
-    parent_dir = os.path.join(tempfile.gettempdir(), "iree", "modules")
+  parent_dirs = [
+      FLAGS.artifacts_dir,
+      os.environ['TEST_UNDECLARED_OUTPUTS_DIR'],
+      os.path.join(tempfile.gettempdir(), "iree", "modules"),
```

Comment:
Do you know where outputs in `TEST_TMPDIR` get symlinked to after the test invocation? It's temporarily saved here (`/dev/shm/bazel-sandbox.19090a49aee87113714d6b0f1be1774a673978988ab65419246c1db550a7e52e/linux-sandbox/414/execroot/iree_core/_tmp/98d3b70476c9b783e03aa6eaceb9203a`) on my system, but I can't find any information in their documentation.

My thinking for not including `TEST_TMPDIR` in the fallbacks was that if `TEST_UNDECLARED_OUTPUTS_DIR` isn't set then at least the artifacts are put some place reachable. It would be unfortunate if someone ran `get_e2e_artifacts.py` and it failed because `TEST_UNDECLARED_OUTPUTS_DIR` was unset, and they didn't have an obvious way to go forward.

Reply:
It's destroyed after the test invocation. That's kind of the whole point :-D Tests should be hermetic and not fiddling with random parts of the filesystem. If someone wants to save the artifacts they can still specify `--artifacts-dir`

Reply:
Okay, SGTM!